### PR TITLE
ApiServiceCollectionExtensions.AddProxyAddresses: Add more tests

### PR DESCRIPTION
**Pull request by Bond-009**

### CODE REVIEW

**Review by barronpm — APPROVED** [review]

File: MediaBrowser.Common/Net/IPHost.cs [hunk 1]
```diff
@@ -410,30 +409,31 @@ private bool ResolveHost()
...
-                    return;
+                    _addresses = Dns.GetHostAddresses(hostName);
                 }
-
-                if (Uri.CheckHostName(HostName).Equals(UriHostNameType.Dns))
+                catch (SocketException ex)
                 {
-                    try
-                    {
-                        IPHostEntry ip = await Dns.GetHostEntryAsync(HostName).ConfigureAwait(false);
```
Comment (cvium) [inline MediaBrowser.Common/Net/IPHost.cs:429]: Why did you change it? Looks to me like they behave differently https://docs.microsoft.com/en-us/dotnet/api/system.net.dns.gethostentryasync?view=net-5.0#System_Net_Dns_GetHostEntryAsync_System_String_

Comment (Bond-009) [reply MediaBrowser.Common/Net/IPHost.cs:429]: https://source.dot.net/#System.Net.NameResolution/System/Net/Dns.cs,409
We only use the addresses.

**Review by cvium — APPROVED** [review]
